qiita_ware/ebi.py

```diff
@@ -143,33 +148,72 @@ def __init__(self, preprocessed_data_id, action):
             self.new_investigation_type = it
         else:
             # This should never happen
-            raise ValueError("Unrecognized investigation type: '%s'. This "
-                             "term is neither one of the official terms nor "
-                             "one of the user-defined terms in the ENA "
-                             "ontology")
-
-        self.ebi_dir = join(qiita_config.working_dir, 'ebi_submission_',
-                            str(preprocessed_data_id))
-        self.sequence_files = []
-        self.study_xml_fp = None
-        self.sample_xml_fp = None
-        self.experiment_xml_fp = None
-        self.run_xml_fp = None
-        self.submission_xml_fp = None
+            error_msgs.append("Unrecognized investigation type: '%s'. This "
+                              "term is neither one of the official terms nor "
+                              "one of the user-defined terms in the ENA "
+                              "ontology." % it)
+
+        self.ebi_dir = join(qiita_config.working_dir,
+                            'ebi_submission_%d' % preprocessed_data_id)
+        get_output_fp = partial(join, self.ebi_dir, 'xml_dir')
+        self.xml_dir = get_output_fp()
+        self.study_xml_fp = get_output_fp('study.xml')
+        self.sample_xml_fp = get_output_fp('sample.xml')
+        self.experiment_xml_fp = get_output_fp('experiment.xml')
+        self.run_xml_fp = get_output_fp('run.xml')
+        self.submission_xml_fp = get_output_fp('submission.xml')
         self.pmids = s.pmids
 
-        # dicts that map investigation_type to library attributes
-        lib_strategies = {'metagenome': 'POOLCLONE',
-                          'mimarks-survey': 'AMPLICON'}
-        lib_selections = {'mimarks-survey': 'PCR'}
-        lib_sources = {}
-
-        self.library_strategy = lib_strategies.get(key, "OTHER")
-        self.library_source = lib_sources.get(key, "METAGENOMIC")
-        self.library_selection = lib_selections.get(key, "unspecified")
-
-        # This will hold the submission's samples, keyed by the sample name
+        # getting the restrictions
+        st_missing = self.sample_template.check_restrictions(
+            [self.sample_template.columns_restrictions['EBI']])
+        pt_missing = self.prep_template.check_restrictions(
+            [self.prep_template.columns_restrictions['EBI']])
+        # testing if there are any missing columns
+        if st_missing:
```

Comment:
Should these errors include the prep template id as the error in line 209?

I'm also noticing that these errors have a common pattern: provide the study, prep template and preprocessed data id and then the error. Can this duplication be reduced by modifying this error messages to just include the error (e.g. her will be
```python
error_msgs.append("Missing column in the sample template: %s" % ', '.join(list(st_missin)))
```
and then on line 214 do something like:
```python
if error_msgs:
    error_msgs = "Errors found during EBI submission for study #%d, preprocessed data #%d and prep template #%d:\n" % (s.id, preprocessed_data_id, self.prep_template.id, '\n'.join(error_msgs))
```
I think it will be easy to maintain and the user will receive less duplicated information